libminifi/src/controllers/SSLContextService.cpp

```diff
@@ -196,16 +196,16 @@ bool SSLContextService::configure_ssl_context(SSL_CTX *ctx) {
   }
 
   // Security level set to 0 for backwards compatibility to support TLS versions below v1.2
-  SSL_CTX_set_security_level(ctx, 0);
+  if (minimum_tls_version_ < TLS1_2_VERSION || maximum_tls_version_ < TLS1_2_VERSION) {
```

Comment:
isn't it enough to check the `minimum_tls_version_`? I assume the value `-1` is "don't care", so if we do not care about the `maximum_tls_version_` event if the `minimum_tls_version_` disallows it we set the security level to 0

Reply:
By default only TLS 1.2 or above is enabled in OpenSSL3. If we set the minimum version or the maximum below that, we should enable legacy mode to allow versions older than 1.2 to be used after version negotiation. But thinking about what you said, I think the `-1` value should also be checked, because if we "do not care" then we should better use the default values which is allow the TLS 1.2+ versions only. Legacy versions should only be used when specified. Updated in 234608e9d16a8e2c7d60b049fb2264bcaaf1c2fd

Reply:
I'm still having wrapping my head around this, so for the negotiation to be able to chose < 1.2 we have to set the security level to 0, but we also have to either specify a minimum version < 1.2 or don't set the minimum version at all, so as I understand the < 1.2 can only NOT be negotiated if `minimum_tls_version_ >= 1.2` all other cases allow for < 1.2 but for that we need to also set the security level to 0, so the condition should only be `minimum_tls_version_ != -1 && minimum_tls_version_ < TLS1_2_VERSION`, so the user explicitly set the minimum version to allow pre-1.2, it seems to me that the maximum version does not really play a role

Reply:
I see, so `-1` is not "don't care" but the default